### DISCUSSION TITLE
Amplifier placement queries

### DISCUSSION THREAD

**Discussion by SaiBhyri** [Telecominfraproject/oopt-gnpy#493] [Q&A]
Hi Team,

- What is the logic for Raman amp placement?
- Is there a mechanism to tell GNPy to consider Raman amp on a certain span or forcing the Raman amp on the specific span only option?
- Is the possible to define amp placement rules in GNPy based on span loss?
- What is the function in code which does amp placement?

Thanks,
Sai

**Comment by EstherLerouzic** [comment]
Dear Sai,

The current autodesign does not consider Raman + EDFA hybrid amplifier differently from EDFA: it selects the best amp among those that are allowed for design ('allowed_for_design': true) and that satisfy gain and power constraint. eligible amps are designated by the "allowed_for_design" true in the eqpt library. the selection function checks that there are candidates without the need for padding (there exist a candidate whose gain min  is above target gain) else it raises an error.

You can use the variety_list attribute option in Edfa instance to specify a list of amps:
`
    {
      "uid": "east edfa in SITE1 to ILA1",
      "type": "Edfa",
      "variety_list": ["std_medium_gain", "std_low_gain", "std_fixed_gain"],
      "operational": {
        "delta_p": 0
      },
      "metadata": {
        "location": {
          "latitude": 2.0,
          "longitude": 0.0,
          "city": "SITE1",
          "region": "RLD"
        }
      }
    },
`
- There is no automatic decision on where to place amplifier yet. The autodesign uses the site designated as amp sites by the user.
- The autodesign code is in "selecte_edfa and set_egress_amplifier in gnpy.core.network.py

**Comment by SaiBhyri** [comment]
Thanks Esther for your response.
Does it select the amp which gives best OSNR or just the first one in the list that satisfies gain and power constraint?

Regards,
Sai

**Reply by EstherLerouzic** [reply]
it selects the one with the best NF in this list,

regards
Esther

**Reply by SaiBhyri** [reply]
Thanks Esther
